### DISCUSSION TITLE
Argocd without crd

### DISCUSSION THREAD

**Discussion by Rez0k** [argoproj/argo-cd#6165] [Q&A]
Hi,
I am trying to install argocd on my cluster. is it possible to install argo cd without crds (namespace privileges only).

And another question:
I want to add another cluster to argo cd but when I am trying to add i get an error saying argocd-manager-role require permissions that I don't have.
How can I modify argocd manager role to ask for minimum permissions before using the argocd cluster add command?

Thanks!

**Comment by wtam2018** [comment]
@Rez0k you may want to try the `namespace-install.yaml` manifest.   https://github.com/argoproj/argo-cd/blob/master/manifests/README.md

**Reply by Rez0k** [reply]
There is a note in the namespace install section:
Note: Argo CD CRDs are not included into namespace-install.yaml. and have to be installed separately. The CRD manifests are located in manifests/crds directory. Use the following command to install them:

Can I bypass the crds install?

**Comment by jannfis** [comment]
Hi,

> is it possible to install argo cd without crds

The cluster you install Argo CD to needs to provide the CRDs, because they are a vital piece of everything Argo CD does. If you don't have the required permissions, you should ask the cluster administrator to install them, and also to grant you full control to `Application` and `AppProject` resources in your namespace. It is not possible to run Argo CD without the CRDs being available.

> How can I modify argocd manager role to ask for minimum permissions before using the argocd cluster add command?

You can manually create a `ServiceAccount` and appropriate `Role` and `RoleBinding` (or corresponding cluster level resources) on the remote cluster as needed, and then connect the cluster by creating an appropriate cluster secret in the Argo CD installation namespace as described [here](https://argo-cd.readthedocs.io/en/release-1.8/operator-manual/declarative-setup/#clusters). The `argocd cluster add` command currently only creates full cluster-level admin roles on the target cluster.

HTH.

**Reply by Rez0k** [reply]
Thanks al lot!

I managed to connect to the cluster with argo the problem is that when i am trying to sync my project I get a lot of errors saying missing permissions.
Is there a way to make argo demand less permission? I am sure argo dont need each and every one of this permissions.
(I already edited the argocd-cluster-controller and argocd-server cluster roles)
What I missed?

:)

**Reply by Rez0k** [reply]
argocd-application*-controller

**Reply by jannfis** [reply]
You might want to restrict the resources that Argo CD tries to access to match your actual permissions, as described [here](https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#resource-exclusioninclusion)

**Reply by Rez0k** [reply]
Thanks!
